### PR TITLE
Extract repo preparation to a script

### PR DESCRIPTION
**Pull request by ChALkeR**
Fetches specific subdep hashes instead of master, also verifies boost shasum automatically.

This also drops all the complex submodule updates in favor of this simple build script.

The aim is to ensure that the build is always done using specific dependency versions.

### CODE REVIEW

File: prepare.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+if [[ ! -f "boost_1_69_0.tar.gz" ]]; then
+  curl -LO "${boost_url}"
+fi
+if [ "$(shasum -a 256 'boost_1_69_0.tar.gz' | sed s/' .*'//)" != "${boost_sha256}" ]; then
+  echo "Invalid boost shasum!"
+  exit -1
+fi
+
+echo "Extracting boost..."
+rm -rf 'contrib/boost-sd' && mkdir -p 'contrib/boost-sdk'
```
Comment (Sekhmet) [inline prepare.sh]:
```suggestion
rm -rf 'contrib/boost-sdk' && mkdir -p 'contrib/boost-sdk'
```

File: prepare.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
```
Comment (Sekhmet) [inline prepare.sh]:
```suggestion
#!/usr/bin/env bash
```

It's not entirely POSIX compatible.

File: prepare.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+monero_core_custom_url='https://github.com/ExodusMovement/monero-core-custom'
+monero_core_custom_hash='c601fdc3a7aa0c449a3e2c99df230f503fb67e3c'
+
+## Boost, hash should match upstream documented
+boost_url='https://dl.bintray.com/boostorg/release/1.69.0/source/boost_1_69_0.tar.gz'
+boost_sha256='9a2c2819310839ea373f42d69e733c339b4e9a19deab6bfec448281554aa4dbb' # per https://www.boost.org/users/history/version_1_69_0.html
+
+if [ "$(basename "$(pwd)")" != "mymonero-core-js" ]; then
+  echo "Should be ran from the repo dir!"
+  exit -1
```
Comment (Sekhmet) [inline prepare.sh]:
```suggestion
  exit 1
```

Should return 0-255. There are similar `exit` calls below.

File: prepare.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+fi
+
+function clonerepo { # source, target, commit
+  rm -rf "$2"
+  oldpwd="$(pwd)"
+
+  git clone "${1}" "${2}" || exit -1
```
Comment (Sekhmet) [inline prepare.sh]: We don't need to `exit` there unless we want specific exit code.

File: prepare.sh (hunk 1 above)
Comment (ChALkeR) [reply prepare.sh]: Oh. Thanks.

File: prepare.sh (hunk 4 above)
Comment (ChALkeR) [reply prepare.sh]: We do, otherwise the script will continue to execute even if clone failed.

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**